File: app/src/main/java/biz/dealnote/messenger/mvp/view/IChatView.java
package biz.dealnote.messenger.mvp.view;

import android.net.Uri;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.ArrayList;
import java.util.List;

import biz.dealnote.messenger.crypt.KeyLocationPolicy;
import biz.dealnote.messenger.model.LoadMoreState;
import biz.dealnote.messenger.model.Message;
import biz.dealnote.messenger.model.ModelsBundle;
import biz.dealnote.messenger.model.Peer;
import biz.dealnote.messenger.upload.UploadDestination;

/**
 * Created by ruslan.kolbasa on 05.10.2016.
 * phoenix
 */
public interface IChatView extends IBasicMessageListView, IErrorView {

    void setupLoadUpHeaderState(@LoadMoreState int state);
    void displayDraftMessageAttachmentsCount(int count);
    void displayDraftMessageText(String text);
    void setupSendButton(boolean canSendNormalMessage, boolean canSendVoiceMessage);
    void displayToolbarTitle(String text);
    void displayToolbarSubtitle(String text);
    void setRecordModeActive(boolean active);
    void requestRecordPermissions();
    void displayRecordingDuration(long time);
    void doCloseAfterSend();

    void goToMessageAttachmentsEditor(int accountId, int messageOwnerId, @NonNull UploadDestination destination,
                                      String body, @Nullable ModelsBundle attachments);
    void showErrorSendDialog(@NonNull Message message);
    void notifyItemRemoved(int position);

    void configOptionMenu(boolean canLeaveChat, boolean canChangeTitle, boolean canShowMembers,
                          boolean encryptionStatusVisible, boolean encryprionEnabled, boolean encryptionPlusEnabled, boolean keyExchangeVisible);
    void goToSearchMessage(int accountId, @NonNull Peer peer);
    void showImageSizeSelectDialog(@NonNull List<Uri> streams);

    void resetUploadImages();
    void resetInputAttachments();
    void notifyChatResume(int accountId, int peerId, String title, String image);
    void goToConversationAttachments(int accountId, int peerId);
    void goToChatMembers(int accountId, int chatId);
    void showChatTitleChangeDialog(String initialValue);
    void forwardMessagesToAnotherConversation(@NonNull ArrayList<Message> messages, int accountId);
    void diplayForwardTypeSelectDialog(@NonNull ArrayList<Message> messages);
    void setEmptyTextVisible(boolean visible);
    void setupRecordPauseButton(boolean avilable, boolean isPlaying);
    void displayIniciateKeyExchangeQuestion(@KeyLocationPolicy int keyStoragePolicy);
    void showEncryptionKeysPolicyChooseDialog(int requestCode);
    void showEncryptionDisclaimerDialog(int requestCode);
}

File: app/src/main/java/biz/dealnote/messenger/mvp/view/IChatView.kt
package biz.dealnote.messenger.mvp.view

import android.net.Uri
import biz.dealnote.messenger.crypt.KeyLocationPolicy
import biz.dealnote.messenger.model.*
import biz.dealnote.messenger.upload.UploadDestination

/**
 * Created by ruslan.kolbasa on 05.10.2016.
 * phoenix
 */
interface IChatView : IBasicMessageListView, IErrorView {

    fun setupLoadUpHeaderState(@LoadMoreState state: Int)
    fun displayDraftMessageAttachmentsCount(count: Int)
    fun displayDraftMessageText(text: String?)
    fun displayToolbarTitle(text: String?)
    fun displayToolbarSubtitle(text: String?)
    fun requestRecordPermissions()
    fun displayRecordingDuration(time: Long)
    fun doCloseAfterSend()

    fun setupPrimaryButtonAsEditing(canSave: Boolean)
    fun setupPrimaryButtonAsRecording()
    fun setupPrimaryButtonAsRegular(canSend: Boolean, canStartRecoring: Boolean)

    fun displayPinnedMessage(pinned: Message?, canChange: Boolean)
    fun hideInputView()

    fun goToMessageAttachmentsEditor(accountId: Int, messageOwnerId: Int, destination: UploadDestination,
                                     body: String?, attachments: ModelsBundle?)

    fun showErrorSendDialog(message: Message)
    fun notifyItemRemoved(position: Int)

    fun configOptionMenu(canLeaveChat: Boolean, canChangeTitle: Boolean, canShowMembers: Boolean,
                         encryptionStatusVisible: Boolean, encryprionEnabled: Boolean, encryptionPlusEnabled: Boolean, keyExchangeVisible: Boolean)

    fun goToSearchMessage(accountId: Int, peer: Peer)
    fun showImageSizeSelectDialog(streams: List<Uri>)

    fun resetUploadImages()
    fun resetInputAttachments()
    fun notifyChatResume(accountId: Int, peerId: Int, title: String?, image: String?)
    fun goToConversationAttachments(accountId: Int, peerId: Int)
    fun goToChatMembers(accountId: Int, chatId: Int)
    fun showChatTitleChangeDialog(initialValue: String?)
    fun forwardMessagesToAnotherConversation(messages: ArrayList<Message>, accountId: Int)
    fun diplayForwardTypeSelectDialog(messages: ArrayList<Message>)
    fun setEmptyTextVisible(visible: Boolean)
    fun setupRecordPauseButton(available: Boolean, isPlaying: Boolean)
    fun displayIniciateKeyExchangeQuestion(@KeyLocationPolicy keyStoragePolicy: Int)
    fun showEncryptionKeysPolicyChooseDialog(requestCode: Int)
    fun showEncryptionDisclaimerDialog(requestCode: Int)
    fun showEditAttachmentsDialog(attachments: MutableList<AttachmenEntry>)

    fun displayEditingMessage(message: Message?)

    fun notifyEditAttachmentChanged(index: Int)
    fun notifyEditAttachmentRemoved(index: Int)
    fun startImagesSelection(accountId: Int, ownerId: Int)
    fun notifyEditAttachmentsAdded(position: Int, size: Int)
    fun notifyEditUploadProgressUpdate(index: Int, progress: Int)
    fun startVideoSelection(accountId: Int, ownerId: Int)
    fun startDocSelection(accountId: Int, ownerId: Int)
    fun startCamera(fileUri: Uri)
    fun showDeleteForAllDialog(ids: ArrayList<Int>)
}